lib/pure/parseutils.nim

```diff
@@ -64,104 +64,158 @@ const
 proc toLower(c: char): char {.inline.} =
   result = if c in {'A'..'Z'}: chr(ord(c)-ord('A')+ord('a')) else: c
 
-proc parseHex*(s: string, number: var int, start = 0; maxLen = 0): int {.
-  rtl, extern: "npuParseHex", noSideEffect.}  =
-  ## Parses a hexadecimal number and stores its value in ``number``.
-  ##
-  ## Returns the number of the parsed characters or 0 in case of an error. This
-  ## proc is sensitive to the already existing value of ``number`` and will
-  ## likely not do what you want unless you make sure ``number`` is zero. You
-  ## can use this feature to *chain* calls, though the result int will quickly
-  ## overflow.
-  ##
-  ## If ``maxLen == 0`` the length of the hexadecimal number has no upper bound.
-  ## Else no more than ``start + maxLen`` characters are parsed, up to the
-  ## length of the string.
-  runnableExamples:
-    var value = 0
-    discard parseHex("0x38", value)
-    assert value == 56
-    discard parseHex("0x34", value)
-    assert value == 56 * 256 + 52
-    value = -1
-    discard parseHex("0x38", value)
-    assert value == -200
+template parseNum[T: SomeInteger](base: string, s: string, number: var T, start = 0, maxLen = 0): untyped =
```

Comment:
To be consistent with the body, `base` should be a `static string`.

Also in this case an enum switch is preferable to enforce compiler check of the passed value.
Even if the code is quite simple, Nim stdlib should set best practices when possible.

All in all I think generalizing the base (hex, oct, bin) + the result var doesn't seem worth it compared to just the result var